src/cc65/declare.c

```diff
@@ -767,9 +767,9 @@ static int ParseFieldWidth (Declaration* Decl)
     /* TODO: This can be relaxed to be any integral type, but
     ** ParseStructInit currently only supports up to int.
     */
-    if (SizeOf (Decl->Type) != SizeOf (type_uint)) {
+    if (SizeOf (Decl->Type) > SizeOf (type_uint)) {
         /* Only int sized types may be used for bit-fields for now */
-        Error ("cc65 currently only supports unsigned int bit-fields");
+        Error ("cc65 currently only supports char and int bit-fields");
```

Comment:
I think that it should say, `"cc65 currently supports only char-sized and int-sized bit-fields"`.  (We don't want to discourage people from using `short`.)

Reply:
Updated, but note that `short` bit-fields are treated as `unsigned short` like happens with int. Not sure if this is considered a bug since we're already in undefined extension territory.

Reply:
Is every extended bit-field type (including `long`) declared without an `unsigned` treated as `unsigned`? It'll be OK if that's consistent.

Reply:
`long` isn't currently supported, but the most obvious way to add it would make `long` bitfields unsigned. So, I agree, not a bug.